lightning-background-processor/src/lib.rs

```diff
@@ -207,15 +220,15 @@ macro_rules! define_run_body {
 	($persister: ident, $chain_monitor: ident, $process_chain_monitor_events: expr,
 	 $channel_manager: ident, $process_channel_manager_events: expr,
 	 $gossip_sync: ident, $peer_manager: ident, $logger: ident, $scorer: ident,
-	 $loop_exit_check: expr, $await: expr)
+	 $loop_exit_check: expr, $await: expr, $get_timer: expr, $timer_elapsed: expr)
```

Comment:
What necessitates that this method be a macro? Could it be a lambda that also takes lambdas for the process_events, loop_exit, and timer arguments?